File: src/main/java/net/xaviersala/Vaca/Model/Camio.java
package net.xaviersala.Vaca.Model;

import java.util.ArrayList;
import java.util.List;

public class Camio {
	
	
	private double PesMaxim;
    private double PesActual;
    private List<Vaca> Vaques;
	private int MaxVaques;
    private double Litres;
	
    public Camio(double pesMax, int nvaques)
    {
        PesMaxim = pesMax;
        PesActual = 0;
		MaxVaques = nvaques;
        Vaques = new ArrayList<Vaca>();
    }
    
    
    public double getPesMaxim() {
		return PesMaxim;
	}


	public double getPesActual() {
		return PesActual;
	}


	public List<Vaca> getVaques() {
		return Vaques;
	}

	public double getLitres() {
		return Litres;
	}
	
	public int getMaxVaques(){
		return MaxVaques;
	}

	public double GetLitres() { 
    	return Vaques.stream().mapToDouble(i -> i.GetLitres()).sum();
    };

    public boolean EntraVaca(Vaca vaca)
    {
        if (PesActual + vaca.getPes() > PesMaxim && Vaques.size() < MaxVaques)
        {
            return false;
        }

        Vaques.add(vaca);
        PesActual += vaca.getPes();
        return true;
    }

    public boolean TreuVaca(Vaca vaca)
    {
        boolean resultat = Vaques.remove(vaca);
        if (resultat) { 
			PesActual -= vaca.getPes(); 
		}
        return resultat;
    }
	
	public boolean ComprovaVaca(Vaca vaca)
    {
        if (PesActual + vaca.getPes() > PesMaxim && Vaques.size() < MaxVaques)
        {
            return false;
        }

        return true;
    }

    @Override
    public String toString() {
    	return "Camio max: " + PesMaxim + "(" + PesActual + " - vaques: " + Vaques.size() + ")";
    }

}

File: Vaques/Model/Camio.cs
using System.Collections.Generic;
using System.Linq;

namespace Vaques.Model
{
    public class Camio
    {
        public Camio(double pesMax)
        {
            PesMaxim = pesMax;
            _PesActual = 0;
            Vaques = new List<Vaca>();
        }


        public double PesMaxim { get; }
        public double PesActual { get => _PesActual; }
        public List<Vaca> Vaques { get; }

        public double Litres;
        public double _PesActual;
        public double GetLitres() => Vaques.Sum(x => x.GetLitres());

        public bool EntraVaca(Vaca vaca)
        {
            if (PesActual + vaca.Pes > PesMaxim)
            {
                return false;
            }

            Vaques.Add(vaca);
            _PesActual += vaca.Pes;
            return true;
        }

        public bool TreuVaca(Vaca vaca)
        {
            var resultat = Vaques.Remove(vaca);
            if (resultat) { _PesActual -= vaca.Pes; }
            return resultat;
        }

        public override string ToString() => $"Camio max:{PesMaxim} ({PesActual}) - vaques: {Vaques.Count}";
    }
}